parachains/runtimes/collectives/collectives-polkadot/src/weights/frame_system.rs

```diff
@@ -53,80 +52,99 @@ impl<T: frame_system::Config> frame_system::WeightInfo for WeightInfo<T> {
-			.saturating_add(Weight::from_parts(413, 0).saturating_mul(b.into()))
+			.saturating_add(Weight::from_parts(415, 0).saturating_mul(b.into()))
 	}
 	/// The range of component `b` is `[0, 3932160]`.
 	fn remark_with_event(b: u32, ) -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 7_797_000 picoseconds.
-		Weight::from_parts(7_933_000, 0)
+		// Minimum execution time: 7_502_000 picoseconds.
+		Weight::from_parts(7_628_000, 0)
 			.saturating_add(Weight::from_parts(0, 0))
 			// Standard Error: 0
-			.saturating_add(Weight::from_parts(1_455, 0).saturating_mul(b.into()))
+			.saturating_add(Weight::from_parts(1_459, 0).saturating_mul(b.into()))
 	}
-	/// Storage: System Digest (r:1 w:1)
-	/// Proof Skipped: System Digest (max_values: Some(1), max_size: None, mode: Measured)
-	/// Storage: unknown `0x3a686561707061676573` (r:0 w:1)
-	/// Proof Skipped: unknown `0x3a686561707061676573` (r:0 w:1)
+	/// Storage: `System::Digest` (r:1 w:1)
+	/// Proof: `System::Digest` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: UNKNOWN KEY `0x3a686561707061676573` (r:0 w:1)
+	/// Proof: UNKNOWN KEY `0x3a686561707061676573` (r:0 w:1)
 	fn set_heap_pages() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `1485`
-		// Minimum execution time: 4_428_000 picoseconds.
-		Weight::from_parts(4_614_000, 0)
+		// Minimum execution time: 4_097_000 picoseconds.
+		Weight::from_parts(4_294_000, 0)
 			.saturating_add(Weight::from_parts(0, 1485))
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
+	/// Storage: `ParachainSystem::ValidationData` (r:1 w:0)
+	/// Proof: `ParachainSystem::ValidationData` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainSystem::UpgradeRestrictionSignal` (r:1 w:0)
+	/// Proof: `ParachainSystem::UpgradeRestrictionSignal` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainSystem::PendingValidationCode` (r:1 w:1)
+	/// Proof: `ParachainSystem::PendingValidationCode` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainSystem::HostConfiguration` (r:1 w:0)
+	/// Proof: `ParachainSystem::HostConfiguration` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainSystem::NewValidationCode` (r:0 w:1)
+	/// Proof: `ParachainSystem::NewValidationCode` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainSystem::DidSetValidationCode` (r:0 w:1)
+	/// Proof: `ParachainSystem::DidSetValidationCode` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	fn set_code() -> Weight {
-		Weight::from_parts(1_000_000, 0)
+		// Proof Size summary in bytes:
+		//  Measured:  `156`
+		//  Estimated: `1641`
+		// Minimum execution time: 84_469_986_000 picoseconds.
+		Weight::from_parts(87_460_591_000, 0)
```

Comment:
yes, makes sense, because `set_code` was hard-coded,
and just here was added benchmarking for that:
https://github.com/paritytech/substrate/pull/13373
https://github.com/paritytech/substrate/pull/13373

hard-coded value was added here but nobody regenerated till now:
https://github.com/paritytech/cumulus/pull/2547

Reply:
`set_code`'s weight would mostly be in the extrinsic length (like remark). If it's 80bn + the tx length, isn't there a danger that the extrinsic won't be able to fit into a block? (When Async backing lands that's not going to be an issue.)
but given that these are parachains and we don't call `set_code` on parachains to upgrade them it's a moot point.

Reply:
@bkontur can you explain the weight? is it right to use substrate's dev runtime to benchmarks? should not it mostly depend from the length?

Reply:
it should not need the wasm itself to measure the weight, it might only need the len, right?
for this release we at least should make sure `set_code` total weight is still fitting into one block.
the current weight looks wrong

Reply:
@muharem 

> it should not need the wasm itself to measure the weight, it might only need the len, right?

maybe I am missing something, what do you mean by "it might only need the len"? How do you want to write benchmark
for https://github.com/paritytech/substrate/blob/master/frame/system/src/lib.rs#L420-L421? Just with `len`?
```
pub fn set_code(origin: OriginFor<T>, code: Vec<u8>) -> DispatchResultWithPostInfo {
```

> for this release we at least should make sure set_code total weight is still fitting into one block.

How? Do we have already some similar checks for sane weights? Do you want to add some ensurement to:
```
        #[test]
	fn sane() {
          // pseudocode
          assert!(T::BlockMaxWright()::get() >= frame_system::set_code().weights)
        }
```

> the weight value looks wrong

well the weight was calculated with real code, ok, with much bigger kitchensink_runtime, but still calculation went through real code, so what do you think is exactly wrong? should you correct runtime wasm or? What do you suggest?